test/functional/api/v2/discovery_rules_controller_test.rb

```diff
@@ -53,17 +76,120 @@ class Api::V2::DiscoveryRulesControllerTest < ActionController::TestCase
+  end
+
   test "should update taxonomy for discovery rule" do
     rule = FactoryBot.create(:discovery_rule)
     put :update, params: { :id => rule.to_param, :discovery_rule => { :organization_ids => [rule.organizations.first.id] } }
     assert_response :success
   end
 
+  test_attributes :pid => '0f8ec302-f9de-4713-87b7-0f1aca515149'
+  test "should update taxonomies for discovery rule" do
+    min_required_attr = { :name => 'new_rule', :search => 'CPU_Count = 1', :hostgroup_id => hostgroups(:unusual).id }
+    rule = DiscoveryRule.new(min_required_attr)
+    assert rule.save
+    assert rule.organizations.empty?
+    assert rule.locations.empty?
+    hostgroup = FactoryBot.create(:hostgroup, :with_os, :with_rootpass, :organizations => [organization_one], :locations => [location_one])
+    put :update, params: { :id => rule.to_param, :discovery_rule => {
+      :organization_ids => [organization_one.id],
+      :location_ids => [location_one.id],
+      :hostgroup_id => hostgroup.id
+    }}
+    assert_response :success
+    rule.reload
+    assert_equal [organization_one.id], rule.organizations.map {|org| org.id}
+    assert_equal [location_one.id], rule.locations.map {|loc| loc.id}
+  end
+
+  test_attributes :pid => '2c5ecb7e-87bc-4980-9620-7ae00e3f360e'
+  test "should update search rule" do
+    rule = FactoryBot.create(:discovery_rule)
+    new_search = 'Location = Default_Location'
+    put :update, params: { :id => rule.to_param, :discovery_rule => { :search => new_search } }
+    assert_response :success
+    rule.reload
+    assert_equal new_search, rule.search
+  end
+
+  test_attributes :pid => '33084060-2866-46b9-bfab-23d91aea73d8'
+  test "should update host limit" do
+    rule = FactoryBot.create(:discovery_rule)
+    new_max_count = 150
+    assert_not_equal new_max_count, rule.max_count
+    put :update, params: { :id => rule.to_param, :discovery_rule => { :max_count => new_max_count } }
+    assert_response :success
+    rule.reload
+    assert_equal new_max_count, rule.max_count
+  end
+
+  test_attributes :pid => '330aa943-167b-46dd-b434-1a6e5fe8f283'
+  test "test_positive_disable" do
+    rule = FactoryBot.create(:discovery_rule)
+    assert rule.enabled
+    put :update, params: { :id => rule.to_param, :discovery_rule => { :enabled => false } }
+    assert_response :success
+    rule.reload
+    refute rule.enabled
+  end
+
+  test_attributes :pid => '9fdba953-dcc7-4532-9204-17a45b0d9e05'
   test "should destroy discovery rule" do
     rule = FactoryBot.create(:discovery_rule)
     assert_difference('DiscoveryRule.unscoped.count', -1) do
       delete :destroy, params: { :id => rule.to_param }
     end
     assert_response :success
+    assert_raise(ActiveRecord::RecordNotFound) { DiscoveryRule.unscoped.find(rule.id) }
+  end
+
+  test_attributes :pid => '415379b7-0134-40b9-adb1-2fe0adb1ac36'
+  test "should not create with too long name" do
+    assert_difference('DiscoveryRule.unscoped.count', 0) do
```

Comment:
this feels more like a model/unit test, as this mostly tests validations

Reply:
We have no negative test in this controller, think we should leave this one here.

Reply:
:+1: